perfkitbenchmarker/benchmarks/fio_benchmark.py

```diff
@@ -61,53 +155,37 @@ def GetInfo():
 
-     This includes installing fio, bc. and libaio1 and insuring that the
-     attached disk is large enough to support the fio benchmark.
+     This includes installing fio, bc, and libaio1 and insuring that
+     the attached disk is large enough to support the fio
+     benchmark. We also make sure the job file is always located at
+     the same path on the local machine.
 
   Args:
     benchmark_spec: The benchmark specification. Contains all data that is
         required to run the benchmark.
+
   """
-  vms = benchmark_spec.vms
-  vm = vms[0]
+
+  if (FLAGS.fio_jobfile and
+      (FLAGS.against_device or FLAGS.device_fill_size or FLAGS.io_depths)):
+    logging.warning('Fio job file specified. Ignoring other fio options.')
+  if FLAGS.device_fill_size and not FLAGS.against_device:
+    logging.warning('--device_fill_size has no effect without --against_device')
+
+  vm = benchmark_spec.vms[0]
   logging.info('FIO prepare on %s', vm)
   vm.Install('fio')
-  file_path = data.ResourcePath(flags.FLAGS.fio_jobfile)
-
-  disk_size_kb = vm.GetDeviceSizeFromPath(vm.GetScratchDir())
-  amount_memory_kb = vm.total_memory_kb
-  if disk_size_kb < amount_memory_kb * flags.FLAGS.memory_multiple:
-    logging.error('%s must be larger than %dx memory"',
-                  vm.GetScratchDir(),
-                  flags.FLAGS.memory_multiple)
-    # TODO(user): exiting here is probably not the correct behavor.
-    #    When FIO is run across a data set which is too not considerably
-    #    larger than the amount of memory then the benchmark results will be
-    #    invalid. Once the benchmark results are returned from Run() an
-    #    invalid (or is that rather a 'valid' flag should be added.
-    exit(1)
-  if FLAGS.against_device:
-    device_path = vm.scratch_disks[0].GetDevicePath()
-    logging.info('Umount scratch disk on %s at %s', vm, device_path)
-    vm.RemoteCommand('sudo umount %s' % vm.GetScratchDir())
-    logging.info('Fill scratch disk on %s at %s', vm, device_path)
-    command = (
-        ('sudo %s --filename=%s --ioengine=libaio '
-         '--name=fill-device --blocksize=512k --iodepth=64 '
-         '--rw=write --direct=1 --size=%s') %
-        (fio.FIO_PATH, device_path, FLAGS.device_fill_size))
-    vm.RemoteCommand(command)
-    logging.info('Removing directory and filename in job file.')
-    with open(data.ResourcePath(flags.FLAGS.fio_jobfile)) as original_f:
-      job_file = original_f.read()
-    job_file = fio.DeleteParameterFromJobFile(job_file, 'directory')
-    job_file = fio.DeleteParameterFromJobFile(job_file, 'filename')
-    tmp_dir = vm_util.GetTempDir()
-    file_path = vm_util.PrependTempDir(FLAGS.fio_jobfile)
-    logging.info('Write modified job file to temp directory %s', tmp_dir)
-    with open(file_path, 'w') as modified_f:
-      modified_f.write(job_file)
-  vm.PushFile(file_path)
+
+  mount_point = vm.scratch_disks[0].mount_point
+  logging.info('Umount scratch disk on %s at %s', vm, mount_point)
```

Comment:
We only umount the device when testing against raw device. If you are creating a fio test file at mount point (/scratch0). The device must be mounted.